### PR TITLE
add new documentation generator; rework advanced tuts into repo examples

### PR DESCRIPTION
**Pull request by compleatang**
One item to note is that the how_to_make_a_service example should be refactored to leverage the docs/test + embedding features of the new documentation generator.

### CODE REVIEW

File: cmd/eris.go [hunk 1]
```diff
@@ -25,14 +25,13 @@ const dmVerMin = version.DM_VER_MIN
 // Defining the root command
 var ErisCmd = &cobra.Command{
 	Use:   "eris COMMAND [FLAG ...]",
-	Short: "The Blockchain Application Platform",
+	Short: "The Ecosystem Application Platform",
 	Long: `Eris is a platform for building, testing, maintaining, and operating
-distributed applications with a blockchain backend. Eris makes it easy
-and simple to wrangle the dragons of smart contract blockchains.
+applications built to run on an ecosystem level.
```
Comment (pietv) [inline cmd/eris.go]: Could you explain this change from `Blockchain` to `Ecosystem`? Where is this term coming from? The person who goes to a site and installs Eris he knows what's he's installing and that this has something to do with blockchains. But for a random searcher (this tagline is used for a short description of both Linux, Mac, and Windows packages the word `Ecosystem` suggests this has something to do with ecology, bioresources, fresh water conservation, etc.) the word `Blockchain` has still need to be added one way or another to a package description. 

So, does `Blockchain Ecosystem Application Platform` sound good for a package description?

Comment (compleatang) [inline cmd/eris.go]: This is part of a larger branding move. https://db.erisindustries.com/eris/2016/06/05/ecosystem-applications/

Comment (pietv) [inline cmd/eris.go]: Okay. What I mean is the word `Ecosystem` doesn't help for a random running of `apt-cache search eris` command.

Comment (compleatang) [inline cmd/eris.go]: good point. in tag listings for package managers blockchains and smart contracts and distributed applications and all the buzz words are fine.

Comment (pietv) [inline cmd/eris.go]: Great. Yes, that's what I mean. For tags (and RPM groups) Blockchains and Smart Contracts are fine, but the software description will no longer include the word "blockchain". Gotcha.

Comment (compleatang) [inline cmd/eris.go]: :+1:

**Comment by pietv** [comment]
LGTM

**Pull request merged**